crates/ruff_linter/src/rules/wemake_python_styleguide/rules/too_short_name.rs

```diff
@@ -0,0 +1,159 @@
+use std::path::Path;
+
+use ruff_python_ast::{Alias, Expr, ExprAttribute, ExprName, Identifier, Parameter};
+use ruff_text_size::{Ranged, TextRange};
+
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+use crate::settings::LinterSettings;
+
+use crate::rules::wemake_python_styleguide::helpers::naming;
+
+/// ## What it does
+/// Checks for short variable or module names.
+///
+/// The length of a variable or module name is determined by its number of characters
+/// excluding leading or trailing underscores.
+///
+/// ## Why is this bad?
+/// It is hard to understand what the variable means and why it is used, if its name is too short.
+///
+/// ## Example
+/// ```python
+/// x = 1
+/// y = 2
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x_coordinate = 1
+/// abscissa = 2
+/// ```
+///
+/// ## Options
+/// - [`wemake-python-styleguide.min-name-length`][wemake-python-styleguide.min-name-length]
+///
+/// [wemake-python-styleguide.min-name-length]: ../../settings#wemake-python-styleguide-min-name-length
+#[violation]
+pub struct TooShortName {
+    pub name: String,
+    pub is_module: bool,
+}
+
+impl Violation for TooShortName {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        let TooShortName { name, is_module } = self;
+        format!(
+            "Found too short {}: `{}`",
+            if *is_module { "module name" } else { "name" },
+            name,
+        )
+    }
+}
+
+pub(crate) struct Checkable(Identifier);
+
+impl From<&Identifier> for Checkable {
+    fn from(value: &Identifier) -> Self {
+        Self(value.clone())
+    }
+}
```

Comment:
Yes it seems expensive. I reworked the code a little bit so that `Checkable` is now an `enum` that works with references to `Identifier`, `Parameter`, `Alias` and `Expr` respectively.